### PR TITLE
Add logging and assertions to increase defensiveness of code

### LINKED ISSUE #334: Add logging to commands and parsers

### LINKED ISSUE #335: Add assertions to commands and parsers

### PR DESCRIPTION
**Pull request by AudreyFelicio**

### CODE REVIEW

File: src/main/java/seedu/stock/logic/commands/ListBookmarkCommand.java [hunk 1]
```diff
@@ -19,11 +22,15 @@
...
 
+    private static final Logger logger = LogsCenter.getLogger(ListBookmarkCommand.class);
+
 
     @Override
     public CommandResult execute(Model model) {
+        logger.log(Level.INFO, "Starting to execute list bookmark command");
         requireNonNull(model);
         model.updateFilteredStockList(IS_BOOKMARKED);
+        logger.log(Level.INFO, "Finished listing all stocks successfully");
```
Comment (Ashley-Lau) [inline src/main/java/seedu/stock/logic/commands/ListBookmarkCommand.java]: Maybe can put a bookmark after all to distinguish against list all command?

File: src/main/java/seedu/stock/logic/commands/ListLowStocksCommand.java [hunk 2]
```diff
@@ -19,11 +22,15 @@
...
 
+    private static final Logger logger = LogsCenter.getLogger(ListLowStocksCommand.class);
+
 
     @Override
     public CommandResult execute(Model model) {
+        logger.log(Level.INFO, "Starting to execute list low stocks command");
         requireNonNull(model);
         model.updateFilteredStockList(IS_LOW_STOCK);
+        logger.log(Level.INFO, "Finished listing all stocks successfully");
```
Comment (Ashley-Lau) [inline src/main/java/seedu/stock/logic/commands/ListLowStocksCommand.java]: same here.

**Review by Ashley-Lau — APPROVED** [review]
Some recommendations, and thanks for adding logging to the classes! LGTM otherwise.

File: src/main/java/seedu/stock/logic/commands/ListBookmarkCommand.java (hunk 1 above)
Comment (AudreyFelicio) [reply src/main/java/seedu/stock/logic/commands/ListBookmarkCommand.java]: I think that's a great idea, thanks for the suggestion! Will change now.

**Pull request merged**